pkg/kv/kvserver/store_snapshot.go

```diff
@@ -1143,10 +1105,12 @@ func (s *Store) receiveSnapshot(
 		log.Infof(ctx, "accepted snapshot reservation for r%d", header.State.Desc.RangeID)
 	}
 
+	comparisonResult := s.getLocalityComparison(ctx,
```

Comment:
nit: I don't think you need the local here, right? You can inline this into `updateCrossLocalityMetricsOnSnapshotRcvd`.